### ISSUE TITLE
 USB Keyboard input is ignored when entering GRUB boot menu

### ISSUE THREAD

**Issue by TomaszAIR** [Dasharo/dasharo-issues#690]
### Device

N/A

### Dasharo version

N/A

### Brief summary

N/A

### How to reproduce

N/A

### Expected behavior

N/A

### Actual behavior

```
Question: I am trying to boot to the latest Dasharo tools suite usb and I get to the grub boot selection screen but keyboard input is ignore
For 1 second before the grub boot screen I see a message flash something about “inaccessible device”
With a path
Anyone see this issue before?
Strange because I’ve tried two keys and disable secure boot
```

[Source](https://matrix.to/#/!QRtGMVOaSYrKiAjFKA:matrix.org/$gsFKpoeI51GhC-YRBlG3akdInzwkFrICwWBpPgADP0A?via=matrix.org&via=nitro.chat&via=puck.moe)

Also DTS booted from ISO burned on a USB stick

**Comment by tym2k1** [comment]
[In the mentioned thread](https://matrix.to/#/!QRtGMVOaSYrKiAjFKA:matrix.org/$gsFKpoeI51GhC-YRBlG3akdInzwkFrICwWBpPgADP0A?via=matrix.org&via=nitro.chat&via=puck.moe) it was later noted that the issue occurred exclusively while using [Rufus](https://matrix.to/#/!QRtGMVOaSYrKiAjFKA:matrix.org/$4NSQ_AeAuRAoRTMeaerLJVKzbkbNR-8Pg00aQ6WcR64?via=matrix.org&via=nitro.chat&via=hackliberty.org) for flashing, and [ceased when switching to Balena Etcher](https://matrix.to/#/!QRtGMVOaSYrKiAjFKA:matrix.org/$1_O0YVP8ndTVQVFQtYyi4DJmiZ4VQszaNso30wU8pNo?via=matrix.org&via=nitro.chat&via=hackliberty.org). Attempts to reproduce the problem by flashing the ISO onto a USB using Rufus within a virtual Windows 10 environment—exploring all MBR/GPT partition configurations and Rufus's methods for ISOHybrid images (ISO; ISO -> ESP; DD)—were unsuccessful. The issue could not be replicated.

**Comment by TomaszAIR** [comment]
@tym2k1 to summarize, you did flash DTS with Rufus and there was no problem with keyboard in GRUB? I do not see this confirmation in your comment.

**Comment by tym2k1** [comment]
@TomaszAIR Yes, the keyboard worked correctly in GRUB. I could switch between the boot menu entries and edit the boot parameters with the `e` key. I didn't test it on any of the MSI platforms though just on two separate Dell laptops that I had under my hand. We could reproduce issue on the exact [mentioned hardware configuration](https://matrix.to/#/!QRtGMVOaSYrKiAjFKA:matrix.org/$PsiMTMwJqqUT1FjveYPIf6XJ8VVAaQB5M0lZ38xER1c?via=matrix.org&via=nitro.chat&via=hackliberty.org).

**Comment by tym2k1** [comment]
Tested it also on an MSI platform with the same results. I couldn't replicate the issue and the keyboard worked fine in every scenario when flashing an image with Rufus (as well as the virtual keyboard provided by PiKVM). It's possible that maybe the issue could occur only with specific keyboards.
```
Model: MSI PRO Z690-A DDR4
CPU: 12th Gen Intel(R) Core(TM) i5-12600K
Keyboard: Dell KB216t
tested version of meta-dts: 1.2.18
```

**Comment by miczyg1** [comment]
`EFI/BOOT/grub.cfg` content:

``` 
serial --unit=0 --speed=115200 --word=8 --parity=no --stop=1
default=boot
timeout=0
menuentry 'boot'{
linux /bzImage root=PARTUUID=9ee55ef3-02 rootwait  rootwait quiet
}
```

Timeout is 0, so basically there is no opportunity to read user input.

**Comment by tym2k1** [comment]
On the image `.iso` image taken from the [Dasharo releases tag v1.2.18](https://github.com/Dasharo/meta-dts/releases/tag/v1.2.18) the content of `EFI/BOOT/grub.cfg` is:

```
# Automatically created by OE
serial --unit=0 --speed=115200 --word=8 --parity=no --stop=1
default=boot
timeout=10

menuentry 'boot'{
linux /bzImage LABEL=boot root=/dev/ram0  quiet 
initrd /initrd
}

menuentry 'install'{
linux /bzImage LABEL=install-efi root=/dev/ram0  quiet 
initrd /initrd
}
```

@miczyg1 did you flash an ISO or is your image from some other source?

**Comment by miczyg1** [comment]
@tym2k1 I used WIC image for comparison. SO it means there are multiple grub.cfg sources producing different files.

**Comment by macpijan** [comment]
Why do we care about ISO images? They are unusable anyway, and we have just disabled them in the recent releases. How about we simply propose to use WIC images (we already do that), if this is related to ISO only?

**Comment by tym2k1** [comment]
I guess that's valid to close this issue for now. I already verified the issue of different grub configs and yes that is the case as `grub.cfg` is different among `.iso` and `wic.gz` but it also needs to be as `.iso` `grub.cfg` [needs to point to ram](https://github.com/Dasharo/dasharo-issues/issues/690#issuecomment-2003892270) so I guess it's just a matter of managing both grub files as the `.iso` one is generated automatically by OE. I think that we can close this for now until we resume work on including `.iso` artifact in the release

**Comment by macpijan** [comment]
>  I think that we can close this for now

Not necessarily close, simply add the label (as you just have) and instruct users with this problem to use WIC images. Once we want/need to go back to ISO images, we will have a set of issues waiting for resolution.

**Comment by miczyg1** [comment]
> I already verified the issue of different grub configs and yes that is the case as grub.cfg is different among .iso and wic.gz but it also needs to be as .iso grub.cfg https://github.com/Dasharo/dasharo-issues/issues/690#issuecomment-2003892270 so I guess it's just a matter of managing both grub files as the .iso one is generated automatically by OE

This doesn't explain different timeout values in these configs.
